connector/keystone/keystone.go

```diff
@@ -316,7 +384,41 @@ func (p *conn) getUser(ctx context.Context, userID string, token string) (*userR
 	return &user, nil
 }
 
-func (p *conn) getUserGroups(ctx context.Context, userID string, token string) ([]string, error) {
+func (p *conn) getTokenInfo(ctx context.Context, token string) (*tokenInfo, error) {
+	// https://developer.openstack.org/api-ref/identity/v3/#password-authentication-with-unscoped-authorization
+	authTokenURL := p.Host + "/v3/auth/tokens"
+	p.Logger.Infof("Fetching Keystone token info: %s", authTokenURL)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, authTokenURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-Auth-Token", token)
+	req.Header.Set("X-Subject-Token", token)
```

Comment:
I don't understand why we are fetching token info using the same token? Dex has it own service credentials which should be used for generating a token which goes into the X-Auth-Token header while the user's token should go into X-Subject-Token.

Reply:
This is the original implementation in the connector, I just moved it 😅